### PR TITLE
refactor and document formatters

### PR DESCRIPTION
**Pull request by ChrOertlin**
## Description

Refactors the formatter logic of the delivery services. 
Adds Documentation to the formatter logic 

The formatting logic was a bit unstructured and difficult to deal with. I tried to add some more logic to it by introducing same name space boundaries, reducing the amount of logic and hopefully a clearer build structure.

The formatting happens on three levels.
individual path names
path names bundled for certain components (case, sample, mutant)
destination (where files should be delivered to, i.e. a generic base path or a customer inbox)

A siimilar logic was applied to the Filemover service, i.e. CustomerInboxFileMover and BaseFileMover inherit from DestinationFileMover

### Added

-

### Changed

- Removed SampleFileNameFormatters - these are now generic PathNameFormatters for either a nested or flat structure to inform on what is being formatted
- Moved Case, Sample and Mutant file formatters under an abstract class ComponentFormatters
- Renamed DeliveryFileFormatter and UploadFileFormatter to CustomerInboxDeliveryFormatter and BaseDeliveryFormatter to inform on their destination

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/services/deliver_files/deliver_files_service/deliver_files_service.py [hunk 1]
```diff
@@ -12,9 +11,14 @@
...
 from cg.services.deliver_files.file_fetcher.models import DeliveryFiles
-from cg.services.deliver_files.file_formatter.abstract import DeliveryFileFormattingService
-from cg.services.deliver_files.file_formatter.models import FormattedFiles
-from cg.services.deliver_files.file_mover.delivery_files_mover import DeliveryFilesMover
+from cg.services.deliver_files.file_formatter.destination.abstract import (
+    DeliveryDestinationFormatter,
+)
+from cg.services.deliver_files.file_formatter.destination.models import FormattedFiles
+from cg.services.deliver_files.file_mover.abstract import DestinationFilesMover
+from cg.services.deliver_files.file_mover.customer_inbox_service import (
```
Comment (diitaz93) [inline cg/services/deliver_files/deliver_files_service/deliver_files_service.py:19]: unused import

File: cg/services/deliver_files/file_formatter/component_files/case_service.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        self.file_manager = file_manager
+
+    def format_files(self, moved_files: list[CaseFile], delivery_path: Path) -> list[FormattedFile]:
+        """Format the case files to deliver and return the formatted files.
+        args:
+            moved_files: The case files to format
+            delivery_path: The path to deliver the files to
+        """
+        LOG.debug("[FORMAT SERVICE] Formatting case files")
+        """Format the case files to deliver and return the formatted files."""
```
Comment (diitaz93) [inline cg/services/deliver_files/file_formatter/component_files/case_service.py:33]: repeated line

**Review by diitaz93 — APPROVED** [review]
👍

File: cg/services/deliver_files/file_formatter/component_files/case_service.py (hunk 2 above)
Comment (ChrOertlin) [inline cg/services/deliver_files/file_formatter/component_files/case_service.py:33]:
```suggestion
```

**Pull request merged**